sql/src/main/java/io/crate/jobs/transport/NodeDisconnectJobMonitorService.java

```diff
@@ -83,16 +92,22 @@ public void onNodeConnected(DiscoveryNode node) {
 
     @Override
     public void onNodeDisconnected(final DiscoveryNode node) {
-        final Collection<JobExecutionContext> contexts = jobContextService.getContextsByCoordinatorNode(node.getId());
+        final Collection<UUID> contexts = jobContextService.getJobIdsByCoordinatorNode(node.getId()).collect(Collectors.toList());
         if (contexts.isEmpty()) {
-            return;
+            // Disconnected node is not a handler node --> kill jobs on all participated nodes
+            contexts.addAll(jobContextService.getJobIdsByParticipatingNodes(node.getId()).collect(Collectors.toList()));
+            KillJobsRequest killJobsRequest = new KillJobsRequest(contexts);
+            killJobsNodeAction.broadcast(killJobsRequest, new ActionListener<KillResponse>() {
+                @Override
+                public void onResponse(KillResponse killResponse) { }
+
+                @Override
+                public void onFailure(Throwable e) {
+                    LOGGER.warn("failed to send kill request to nodes");
+                }
+            }, Arrays.asList(node.getId()));
```

Comment:
@mfussenegger when I put the local node on the "excluded" list the jobs won't get killed and the queries hang **for a certain time** (however it will get killed later). If I send a `KillRequest` also to the `client = current node` the jobs get killed **immediately**.  Maybe you know an answer for this behavior otherwise I would stick with this implementation since a request to a local node don't harm the network.